src/app/projects/project-list/project-list-item.component.html

```diff
@@ -3,10 +3,8 @@
 <td class="col-provider">{{providerHostname}}</td>
 <td class="col-status"><span class="not-available">N/A</span></td>
 <td class="col-actions">
-  <p-splitButton *ngIf="projectUtilsService.getActionsMenuItems(project).length > 0"
-    label="{{projectUtilsService.runAllActionName}}" [model]="projectUtilsService.getActionsMenuItems(project)"
-    (onClick)="projectUtilsService.openActions(projectUtilsService.runAllActionName, project)"
-    (click)="$event.stopPropagation()">
+  <p-splitButton
+    label="{{projectUtilsService.runAllActionName}}" [model]="menuItems">
```

Comment:
Missing previous behaviors:
- Don't show button if no valid stages, like in the case of no valid .wharf-ci.yml file.
- Run all stages when pressing the main button.

```suggestion
  <p-splitButton *ngIf="menuItems.length > 0"
    label="{{projectUtilsService.runAllActionName}}"
    (onClick)="projectUtilsService.openActions(projectUtilsService.runAllActionName, project)"
    [model]="menuItems">
```

Could also change the hack in `onProjectRowClicked` to just check `event.defaultPrevented` and then return, if preventing the default here:
```suggestion
  <p-splitButton *ngIf="menuItems.length > 0"
    label="{{projectUtilsService.runAllActionName}}"
    (onClick)="projectUtilsService.openActions(projectUtilsService.runAllActionName, project)"
    (click)="$event.preventDefault()"
    [model]="menuItems">
```

or keep the `$event.stopPropagation()`, although I'm not sure about convention here.
Either would avoid matching on if the target includes `p-button` class name though,

Reply:
# Default fuctionality
I restored the onClick functionality that was a big miss on my part.

# Element hidden when wharf-file has no build steps
As for the fuction where the element dissapers when the target wharf file doesnt contain steps- i could see why it should be disabled, with a message if the config is invalid. I cant see any reason to remove the button entierly. That would just be confusing. I added a sugestion to use the disable functionality.

# Click handling
As for the `click` event i believe a sound design strategy for us is to try to follow the design principals set by the framework we are using. I dont think they have a supported use case for this for their table component most advanced table use i found is here https://primefaces.org/primeng/table/crud . Perhaps we should just use the click for the individual cells in the table row and not the entier row? 
![image](https://user-images.githubusercontent.com/4283527/168049324-454f06e7-ddff-4904-a533-84f51aec3724.png)
Perhaps this is what @jilleJr meant [here](https://github.com/iver-wharf/wharf-web/issues/145#issuecomment-1124564896)? (Sitting here thinking i get it now. :) )


In this case im not seeing how the `$event.preventDefault()` would solve the issue.

Reply:
I think adding a button, in similar style to the "Run all" button, at the far right would be better.

So instead of trying to wrap an anchor tag around the table row or cells, we make it a dedicated link.

Reply:
> As for the fuction where the element dissapers when the target wharf file doesnt contain steps- i could see why it should be disabled, with a message if the config is invalid. I cant see any reason to remove the button entierly. That would just be confusing. I added a sugestion to use the disable functionality.

Very good point. Graying out/disabling the button instead of not displaying it at all seems preferable.
With tooltip is even better 👍🏻

> In this case im not seeing how the $event.preventDefault() would solve the issue.

It would have solved the click-through issue with the following change, https://github.com/iver-wharf/wharf-web/pull/148#discussion_r871042834.

But a dedicated button/adding it separately to the columns is better than messing around with the events I reckon 👍🏻

Reply:
This is the disabled look now -> 
![image](https://user-images.githubusercontent.com/4283527/168071211-dbb1ecf8-22c3-4fd7-bf63-d0174f0f1c22.png)

Reply:
> I think adding a button, in similar style to the "Run all" button, at the far right would be better.
> 
> So instead of trying to wrap an anchor tag around the table row or cells, we make it a dedicated link.

Click feels natural and looks ~decent code with routerlinks is a bit duped but i think it will do. 
![image](https://user-images.githubusercontent.com/4283527/168071832-b4b6f59b-0cd0-47ce-bef4-ec1e3ab3a3be.png)

I dont like the idea of adding additional buttons as it feel like unessacery cluttering, this works well with how the bar behaved before and solves all exsisting (best i can tell) problems with clickthrough and double triggers.